File: umwix1002/umwix1002-fundamental/src/main/java/com/umwix1002/solution/lab/Lab4/L4Q5.java
package Lab4;

public class L4Q5 {
	public static void main(String[] args) {
		int p1 = 0, p2 = 0, dice; boolean isOneTurn = true;
		while(Math.max(p1, p2) <= 100) {
			dice = (int) (Math.random() * 6 + 1);
			if(isOneTurn) {
				p1 += dice;
				isOneTurn = dice == 6 ? true : false;
			} else {
				p2 += dice;
				isOneTurn = dice == 6 ? false : true;
			}
		}
		System.out.printf("%s win with %d points!\n", (p1 > p2 ? "Player 1" : "Player 2"), Math.max(p1, p2));
	}
} 

File: umwix1002-TypeScript/lab/lab04/l4q5.ts
import {createRandom} from "../utils/random";

async function l4q5(): Promise<void> {

    const random = createRandom();

    function rollDice() {
        return random.nextInteger(1, 6 + 1);
    }

    async function main(): Promise<void> {
        let isP1Turn: boolean = Math.random() > 0.5;
        let p1: number = 0, p2: number = 0;
        while (Math.max(p1, p2) < 100) {
            const diceValue = rollDice();
            if (isP1Turn) {
                p1 += diceValue;
            } else {
                p2 += diceValue;
            }
            console.log(`${isP1Turn ? "P1" : "P2"} rolled ${diceValue}. Current value: ${isP1Turn ? p1 : p2}`);
            // Player with diceValue=6 can roll dice again
            if (diceValue !== 6) {
                isP1Turn = !isP1Turn;
            }
        }
        console.log(`${p1 > p2 ? "P1" : "P2"} win.`)
    }

    await main();
}

l4q5()
    .then(() => process.exit(0))
    .catch(error => {
        console.error(error);
        process.exit(1);
    })